fili-core/src/main/java/com/yahoo/bard/webservice/web/OutstandingRequestToken.java

```diff
@@ -0,0 +1,153 @@
+// Copyright 2017 Yahoo Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.web;
+
+import com.codahale.metrics.Counter;
+import com.codahale.metrics.Meter;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.security.Principal;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * RequestToken for successfully bound request.
+ */
+public class OutstandingRequestToken extends RequestToken {
+    private final String userName;
+    private final AtomicInteger userCount;
+    private final AtomicInteger globalCount;
+    private boolean isBound;
+
+    private static int DISABLED_RATE = -1;
+    private static final Logger LOG = LoggerFactory.getLogger(OutstandingRequestToken.class);
+
+    /**
+     * Bind outstanding request to token, or fail and set to unbound.
+     *
+     * @param user  The user that issued the request.
+     * @param requestLimit  The limit on the amount of requests that user can issue.
+     * @param requestLimitGlobal  The limit on the amount of requests any user can issue.
+     * @param userCount  The amount of in-flight requests the user hass issued.
+     * @param globalCount  The total amount of in-flight requests.
+     * @param requestMeter  A meter counting the total amount of accepted requests made.
+     * @param rejectMeter  A meter counting the total number of rejected requests made.
+     * @param requestGlobalCounter  A counter tracking the total number of accepted requests made.
```

Comment:
Should the Meter and Counter classes be managed by the RequestTokens or the RateLimiter that creates the tokens? I think the latter makes more sense.